From: Ben Hutchings <ben@decadent.org.uk>
Date: Sun, 13 Jul 2014 20:08:04 +0100
Subject: btrfs: Work around compiler bug on ARM

With gcc 4.6 in wheezy-backports, we see this build failure on armhf:

/«PKGBUILDDIR»/fs/btrfs/scrub.c: In function 'scrub_enumerate_chunks':
/«PKGBUILDDIR»/arch/arm/include/asm/atomic.h:253:2: error: 'asm' operand requires impossible reload

gcc 4.8 in sid is fine.

Turning down the optimisation level to -O1 for this file results in
successful compilation.

--- a/fs/btrfs/Makefile
+++ b/fs/btrfs/Makefile
@@ -17,3 +17,8 @@ btrfs-$(CONFIG_BTRFS_FS_CHECK_INTEGRITY)
 btrfs-$(CONFIG_BTRFS_FS_RUN_SANITY_TESTS) += tests/free-space-tests.o \
 	tests/extent-buffer-tests.o tests/btrfs-tests.o \
 	tests/extent-io-tests.o tests/inode-tests.o tests/qgroup-tests.o
+
+# bwh: Fails to build with gcc-4.6 -O2
+ifdef CONFIG_ARM
+CFLAGS_scrub.o += -O1
+endif
